src/controllers/portfolio/portfolio.ts

```diff
@@ -596,6 +600,21 @@ export class PortfolioController extends EventEmitter {
 
         const forceUpdate = opts?.forceUpdate || areAccountOpsChanged
 
+        // Pass in learnedTokens as additionalHints only on areAccountOpsChanged
+        const fallbackHints = (storagePreviousHints?.fromExternalAPI &&
+          storagePreviousHints?.fromExternalAPI[key]) ?? {
+          erc20s: [],
+          erc721s: {}
+        }
+        const additionalHints =
+          ((areAccountOpsChanged || forceUpdate) &&
```

Comment:
`areAccountOpsChanged` is included in `forceUpdate` but we can keep it if you think it's better just in case